newlib/libc/tinystdio/filestrputalloc.c

```diff
@@ -41,14 +41,14 @@ __file_str_put_alloc(char c, FILE *stream)
 	struct __file_str *sstream = (struct __file_str *) stream;
 	if (sstream->pos == sstream->end) {
                 size_t old_size = sstream->size;
-                char *old = sstream->end - old_size;
+                char *old = POINTER_MINUS(sstream->end, old_size);
```

Comment:
In the past I've silenced these kinds of UBSan errors using something like:

`char *old = old_size == 0 ? sstream->end : sstream->end - old_size;`

The compiler will remove the ternary and fold this to an simple subtraction but it avoids the ubsan error.

Reply:
Hmm never mind, looks like it's only clang that folds it, GCC keeps the check: https://godbolt.org/z/e56roMn13